src/cdk/portal/dom-portal-outlet.ts

```diff
@@ -93,6 +93,28 @@ export class DomPortalOutlet extends BasePortalOutlet {
     return viewRef;
   }
 
+  /**
+   * Attaches a DOM portal by transferring its content into the outlet.
+   * @param portal Portal to be attached.
+   */
+  attachDomPortal(portal: DomPortal) {
+    // Note that we need to convert this into an array, because `childNodes`
+    // is a live collection which will be updated as we add/remove nodes.
+    let transferredNodes = Array.from(portal.element.childNodes);
```

Comment:
When you attach the `DomPortal` you are moving the child elements of provided `Element` / `ElementRef.nativeElement`. I would expect to move the `Element` / `ElementRef.nativeElement` and not its children. Of course this will require placing some hook at `Element`'s original position. With this implementation when someone needs to show some content in `DomPortal`, the content should be placed in some container, e.g. `div`. Then one should tell to `DomPortal` show this `div`. Finally the `DomPortal` will show the child element/elements of the `div`.